winit-core/src/event.rs

```diff
@@ -683,7 +665,7 @@ pub struct RawKeyEvent {
 
 /// Describes a keyboard input targeting a window.
 #[derive(Debug, Clone, Eq, PartialEq, Hash)]
-pub struct KeyEvent {
+pub struct KeyEvent<Extra> {
```

Comment:
I dislike this design, and think we need to discuss further how to do this sort of thing. See also https://github.com/rust-windowing/winit/issues/3064.